src/trans_rdma.c

```diff
@@ -175,6 +175,48 @@ void __attribute__ ((destructor)) msk_internals_fini(void) {
 	}
 }
 
+/**
+ * Reset global state.
+ * Should (only) be called to reset global data in child process after a fork.
+ */
+void msk_lib_reset(void)
+{
+	/* Close epoll fd's */
+	if (msk_global_state->cm_epollfd != 0)
+		close(msk_global_state->cm_epollfd);
+
+	if (msk_global_state->cq_epollfd != 0)
+		close(msk_global_state->cq_epollfd);
+
+	if (msk_global_state->stats_epollfd != 0)
+		close(msk_global_state->stats_epollfd);
+
+	/* Free worker pool resources */
+	if (msk_global_state->worker_pool.w_efd != 0)
+		close(msk_global_state->worker_pool.w_efd);
+
+	if (msk_global_state->worker_pool.m_efd != 0)
+		close(msk_global_state->worker_pool.m_efd);
+
+	if (msk_global_state->worker_pool.thrids)
+		free(msk_global_state->worker_pool.thrids);
+
+	if (msk_global_state->worker_pool.wd_queue)
+		free(msk_global_state->worker_pool.wd_queue);
+
+	/* Brutal re-initialization of global state */
+	memset(msk_global_state, 0, sizeof(*msk_global_state));
+
+	msk_global_state->run_threads = 0;
```

Comment:
I think we should do that beforefreeing the various items/closing fd and try to join threads like in `msk_internals_fini`

Actually we might be able to reuse these directly, it will free/realloc msk_global_state but the cost is negligible compared to thread joins etc so just something like `msk_internals_fini(); rdma_lib_reset(); msk_internals_init();` is probably cleanest?